src/components/TokenBar.js

```diff
@@ -124,13 +124,23 @@ class TokenBar extends React.Component {
   }
 
   render() {
-    const unknownTokens = this.getUnknownTokens();
+    const shouldHideZeroBalanceTokens = wallet.areZeroBalanceTokensHidden();
+    const unknownTokens = this.getUnknownTokens(shouldHideZeroBalanceTokens);
 
     const renderTokens = () => {
       return this.props.registeredTokens.map((token) => {
+        const tokenUid = token.uid;
+        const isTokenHTR = tokenUid === hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
+        const tokenBalance = this.getTokenBalance(tokenUid);
+
+        // Skip every token without balance, except HTR, if the hiding flag is active
+        if (shouldHideZeroBalanceTokens && !isTokenHTR && tokenBalance === '0.00') {
```

Comment:
This probably won't show new registered tokens with zero balance, as described in the project doc.